common/src/main/java/com/microsoft/identity/common/internal/util/ThreadUtils.java

```diff
@@ -50,4 +59,52 @@ public static void sleepSafely(final int sleepTimeInMs, @NonNull final String ta
             }
         }
     }
+
+    /**
+     * Construct a thread pool with specified name and optionally bounded size.
+     * @param corePool The smallest number of threads to keep alive in the pool.
+     * @param maxPool The maximum number of threads to allow in the thread pool, after which RejectedExecutionException will occur.
+     * @param queueSize The number of items to keep in the queue.  If this is < 0, the queue is unbounded.
+     * @param keepAliveTime The amount of time to keep excess (greater than corePool size) threads alive before terminating them.
+     * @param keepAliveUnit The time unit on that time.
+     * @param poolName The name of the thread pool in use.
+     * @return An executor service with the specified properties.
+     */
+    public static ExecutorService getNamedThreadPoolExecutor(final int corePool, final int maxPool, final int queueSize, final long keepAliveTime, @NonNull final TimeUnit keepAliveUnit, @NonNull final String poolName) {
+        final SecurityManager securityManager = System.getSecurityManager();
+        if (queueSize > 0) {
+            return new ThreadPoolExecutor(corePool, maxPool, keepAliveTime, keepAliveUnit,
+                    new ArrayBlockingQueue<Runnable>(queueSize), getNamedThreadFactory(poolName, securityManager));
+        } else if (queueSize == 0) {
+            return new ThreadPoolExecutor(corePool, maxPool, keepAliveTime, keepAliveUnit, new SynchronousQueue<Runnable>(), getNamedThreadFactory(poolName, securityManager));
+        } else { // (queueSize < 0)
+            return new ThreadPoolExecutor(corePool, maxPool, keepAliveTime, keepAliveUnit, new LinkedBlockingQueue<Runnable>(), getNamedThreadFactory(poolName, securityManager));
+        }
+    }
+
+    //Nice thought, but if you're using executors, you're using ThreadGroup whether you want to or not.
+    @SuppressWarnings("PMD.AvoidThreadGroup")
+    private static ThreadFactory getNamedThreadFactory(@NonNull final String poolName, final SecurityManager securityManager) {
```

Comment:
Do we need to make an API affordance for `SecurityManager` here? These are not supported on Android

Reply:
System.getSecurityManager() will return null in this case, and everything will work.